src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1018,10 +1018,16 @@ private void CleanUp()
 
             try
             {
-                // Dispose the process if it's already created
                 if (_nativeProcess != null)
                 {
+                    // on Unix, we need to kill the process to ensure it terminates as Dispose() merely
+                    // closes the redirected streams and the processs does not exit on macOS.  However,
+                    // on Windows, a winexe like notepad should continue running so we don't want to kill it.
+#if UNIX
+                    _nativeProcess.Kill();
+#else
                     _nativeProcess.Dispose();
+#endif
```

Comment:
We still need to Dispose:
```suggestion
#if UNIX
                    _nativeProcess.Kill();
#endif
                    _nativeProcess.Dispose();
```